deploy/cloud/modules/gcp/vm/main.tf

```diff
@@ -0,0 +1,49 @@
+resource "random_id" "id" {
+  byte_length = 4
+}
+
+locals {
+  vm_private_key_file = "${path.module}/gcp-vm-${random_id.id.hex}.pem"
+  vm_username         = "ubuntu"
+  deploy_name         = "${var.deployment_name}-${random_id.id.hex}"
+}
+
+resource "tls_private_key" "gcp_vm_key" {
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+resource "local_file" "ssh_private_key" {
+  filename        = local.vm_private_key_file
+  content         = tls_private_key.gcp_vm_key.private_key_pem
+  file_permission = 0400
+}
+
+resource "google_compute_instance" "vm_instance" {
+  name         = local.deploy_name
+  machine_type = var.machine_type
+  zone         = var.zone
+  labels       = var.specific_tags
+
+  boot_disk {
+    initialize_params {
+      image = var.disk_image
+    }
+  }
+
+  network_interface {
+    network = var.network
+
+    access_config {
+      // Ephemeral public IP
```

Comment:
delete?

Reply:
No, `access_config` is required and used in the `output.tf` file when retrieving information from the network interface.
```
output "gcp_vm_ssh_cmd" {
  description = "Use this command to SSH into the GCP VM instance"
  value       = "ssh -i ${local.vm_private_key_file} ${local.vm_username}@${google_compute_instance.vm_instance.network_interface[0].access_config[0].nat_ip}"
}
```